[ReorderableListView] remove extra margin added after picking up the item

nero-angela opened the pull request:
## Description
As you can see under the gif when picking up an item in `ReorderableListView`, the others below the item move slightly downward. this is because the extra margin is added to the selected item after lifting. So I removed the extra margin and modified the related test.

|before|after|
|:--:|:--:|
|![before](https://user-images.githubusercontent.com/26322627/91946550-433a4600-ed3a-11ea-8342-28950d77740e.gif)|![after](https://user-images.githubusercontent.com/26322627/91946420-39b0de00-ed3a-11ea-8424-c286e133ab2f.gif)|

### Golden test image
|reorderable_list_test.vertical.drop_area.png|reorderable_list_test.horizontal.drop_area.png|
|:--:|:--:|
|![reorderable_list_test vertical drop_area](https://user-images.githubusercontent.com/26322627/92949779-9cbb1700-f496-11ea-9b0a-e6d3f82af3b2.png)|![reorderable_list_test horizontal drop_area](https://user-images.githubusercontent.com/26322627/92949775-9b89ea00-f496-11ea-8d47-e4414310b1fb.png)|

## Related Issues
Fixes https://github.com/flutter/flutter/issues/58411

## Tests

I modified the following tests: 
- [properly determines the vertical drop area extents](https://github.com/flutter/flutter/blob/780f5f99044ce7fcbcd868e2cf6878da02022db5/packages/flutter/test/material/reorderable_list_test.dart#L144)
- [properly determines the horizontal drop area extents](https://github.com/flutter/flutter/blob/780f5f99044ce7fcbcd868e2cf6878da02022db5/packages/flutter/test/material/reorderable_list_test.dart#L659)

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

nero-angela committed:
[ReorderableListView] remove additional margin

dkwingsmt commented on packages/flutter/lib/src/material/reorderable_list.dart:
```diff
@@ -207,7 +207,7 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
 
   // The additional margin to place around a computed drop area.
-  static const double _dropAreaMargin = 8.0;
+  static const double _dropAreaMargin = 0.0;
```
This variable should no longer be necessary. Remove to simplify code.

dkwingsmt commented on packages/flutter/test/material/reorderable_list_test.dart:
```diff
@@ -178,8 +178,8 @@ void main() {
-        // The list view pads the drop area by 8dp.
-        const double kDraggingListHeight = 300.0;
+        // The list view pads the drop area by 0dp.
+        const double kDraggingListHeight = 292.0;
```
Combine `kNonDraggingListHeight` and `kDraggingListHeight` to simplify code.

dkwingsmt reviewed (changes requested):
Good change, with small change requests.

Also it'd be better if you can add a golden test to make sure it looks exactly how we want after picking up the item. If you're up for the challenge, refer to https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter or search for `matchesGoldenFile` in existing test files.

nero-angela committed:
[ReorderableListView] remove unused code

nero-angela committed:
[ReorderableListView] Add vertical and horizontal drop area golden

nero-angela commented:
Thank you @dkwingsmt ! I've simplified the unnecessary code and added golden test. The images are in the description. :)
This is my first time making golden file changes, I'm not sure if I performed the golden test correctly 🤔
Could you please take a look?

dkwingsmt commented:
The Linux analyzer seems unhappy. Can you resolve the lint errors?

nero-angela committed:
[ReorderableListView] resolve prefer_const_constructors issue

nero-angela commented:
sorry for missing declaring const. @dkwingsmt I've resolved the lint error.

dkwingsmt reviewed (approved):
LGTM

dkwingsmt commented:
Theoretically the bot should comment that a golden test has been modified... I'll check it out next week.

nero-angela commented:
Okay. Thank you for reviewing. Check it out and tell me if there is anything I need to do.
Have a nice weekend :)

justinmc commented:
@dkwingsmt @nero-angela The golden tests added in this PR failed when I tried to revert a TextField layout PR from just before this: https://github.com/flutter/flutter/pull/66027

Can you guys think of any reason why that would happen?  These new tests have nothing to do with TextField or InputDecoration layout, right?

justinmc commented:
Actually it seems to be happening on many PRs:
https://github.com/flutter/flutter/pull/61981
https://github.com/flutter/flutter/pull/65884

Piinks commented:
These images don't appear to have arrived on the gold dashboard after landing: https://flutter-gold.skia.org/
There is nothing under by blame, and I can't find reference to the tests. Checking the logs now on master.
fyi @kjlubick

Piinks commented:
Looks like our luci recipes have been changed and disabled gold again. Will file an infra ticket.

Piinks commented:
https://github.com/flutter/flutter/issues/64969 for tracking
